be/src/column/datum_convert.cpp

```diff
@@ -18,8 +19,7 @@ Status datum_from_string(TypeInfo* type_info, Datum* dst, const std::string& str
     return Status::OK();
 }
 
-Status datum_from_string(TypeInfo* type_info, Datum* dst, const std::string& str, MemPool* mem_pool) {
-    const auto type = type_info->type();
+Status datum_from_string(TypeInfo* type_info, FieldType type, Datum* dst, const std::string& str, MemPool* mem_pool) {
```

Comment:
It's weird to pass both `TypeInfo` and `FieldType` at the same time.  How about
```
Status datum_from_string(FieldType type, Datum* dst, const std::string& str, MemPool* mme_pool) {
   ScalarTypeInfo* type_info = get_scalar_type_info(type);
   if (type != nullptr) {
        return datum_from_string(type_info, dst, str, mem_pool);
  } else {
        return Status::InvalidArgument("non scalar type");
  }
}
```

Reply:
Decimal Type require precision and scale.